platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -162,23 +161,19 @@ private ResolutionParameters getResolutionParameters(
             resolutionStrategy.isDependencyVerificationEnabled(),
             resolutionStrategy.isFailingOnDynamicVersions(),
             resolutionStrategy.isFailingOnChangingVersions(),
-            failureResolutions
+            failureResolutions,
+            resolutionStrategy.getCachePolicy().asImmutable()
         );
     }
 
-    private static class ConfigurationResolveContext implements ResolveContext {
+    private static class ConfigurationLegacyResolutionParameters implements LegacyResolutionParameters {
```

Comment:
The goal of this work and other work in the area is to make it so we can implement this interface (and `ResolutionParameters`) for non-configuration resolvable things. 

I'm open to other names, and I think this can be improved, but I think this is fine at the moment.